src/HeapOrdering.sol

```diff
@@ -248,13 +250,15 @@ library HeapOrdering {
         uint96 _removedValue
     ) private {
         uint256 index = _heap.indexOf[_id];
-        uint256 accountsLength = _heap.accounts.length;
+        uint256 accountsLength = _heap.arrayLength;
 
         // Swap the last account and the account to remove, then pop it.
-        swap(_heap, index, accountsLength - 1);
-        if (_size == accountsLength) _heap.size = --_size;
-        _heap.accounts.pop();
+        uint256 newArrayLength = accountsLength - 1;
+        swap(_heap, index, newArrayLength);
+        delete _heap.accounts[newArrayLength];
```

Comment:
perhaps add a comment that it is possible because we are taking the length in sync at any time. Because `delete` does not update `accounts`'s length

Reply:
To me this is not clear enough. It's not obvious that `delete` is not updating the array's length. It could for other reasons that `arrayLength` is updated eg the access is cheaper than `array.length`

Reply:
> It's not obvious that delete is not updating the array's length

There is no longer any array

Reply:
is it clearer like this ?